.gitattributes

```diff
@@ -0,0 +1 @@
+* text=auto eol=lf
```

Comment:
Up for debate whether to use wildcard (because what if future non-text files are added) or be explicit about the text types to convert (though then you'll have to remember if future text types are added to also add them here).

Open to suggestions if you can think of a better option...

Reply:
Yep, it was complaining it was CRLF instead of LF.  If you have `core.autocrlf=true`, Git checks out with CRLF but commits with LF. Because that conversion happens sometime during the commit, it wouldn't be seen as being converted by the pre-commit hook (or even after it, actually, because the cloned files remain CRLF even after the commit).

I could manually change the file to be LF in my editor to fix this, but that seemed like a kinda painful path for Windows folks.

Reply:
Hmm. That's a bit of a mess. I think I'm OK with your proposed change, but would a different option be to set `* text=auto` and the change the lint rules to allow LF or CRLF?

Reply:
If you change the lint rules to allow both, you probably don't need this at all. But then you'd have to then either double check the line endings before merging PRs, or be OK with your repo potentially having inconsistent line endings.

Reply:
But with text=auto, wouldn't git automatically convert the CRLF to LF upon commit? That's the impression I get from reading https://docs.github.com/en/github/using-git/configuring-git-to-handle-line-endings ("text=auto Git will handle the files in whatever way it thinks is best. This is a good default option.") and https://git-scm.com/docs/gitattributes ("When text is set to "auto", the path is marked for automatic end-of-line conversion. If Git decides that the content is text, its line endings are converted to LF on checkin."

Although I have to say the docs on all of this are disappointingly cryptic and incomplete.

Reply:
My understanding is `text=auto` will convert any _new_ files to LF, but will not affect the local copy
The [doc](https://www.git-scm.com/docs/gitattributes) says
> ...ensure[s] that text files that any contributor introduces to the repository have their line endings normalized...

And that is what I observed with `autocrlf` left `true`.  However, if I do
```
git config core.autocrlf false
unix2dos <whatever file>
git commit -m "I messed up the line endings"
git checkout .
```
it doesn't force the file back to LF when I checkout, because it's not a new file.

Possibly this wasn't a fair test, because I was doing it without a remote.

Reply:
Your experimental results sound right to me. But presumably the file that got committed had LF line endings in the repo. The docs are vague, but my understanding is that with text=auto, checkout will respect your autocrlf settings, but commit will normalize all text files to LF. 

If it works the way it does in my head, I like it a bit better, as it seems slightly safer. But I'm not sure it's worth the time we have spent discussing it and then the time to test it (and especially to undo it if it works differently from how I expect). So I would be OK merging what you have, if that's best. Thoughts?

Reply:
I do think you're right about `text=auto` without specifying a particular line ending results in what "it thinks is best".  But there's actually another setting [core.eol](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreeol) which also impacts this
> Sets the line ending type to use in the working directory for files that are marked as text (either by having the text attribute set, or by having text=auto and Git auto-detecting the contents as text). Alternatives are lf, crlf and native, which uses the platform’s native line ending. The default value is native. See gitattributes[5] for more information on end-of-line conversion. Note that this value is ignored if core.autocrlf is set to true or input.

While I'm admittedly not an expert with these settings, from what I'm reading, what I have will work, but it does have the possibly undesirable side-effect of having LF instead of CRLF on Windows machines.  Though with the current lint rules, that's what's needed.  These days, I'm not sure that's much an issue. Editors handle LF just fine on Windows.

Oh, I was mistaken though about needing to manually specify all text types (as of [2.10](https://github.com/git/git/blob/master/Documentation/RelNotes/2.10.0.txt#L248))

There might be a better approach I haven't thought of yet, but I think merging this would at least be an improvement since at least the files will already be in LF instead of the user having to do the conversion himself/herself.

Reply:
Sounds good. Let's go with what's there. And let's hope someone writes up some comprehensible documentation on all of this someday!